Announce Moving Ruby repository to Git (#2050)
ruby/www.ruby-lang.org: en/news/_posts/2019-04-23-move-to-git-from-svn.md

---
layout: news_post
title: "Move to Git from Subversion"
author: "hsbt"
translator:
date: 2019-04-23 00:00:00 +0000
lang: en
---

The canonical repository of the Ruby language was moved to Git from Subversion.

> The new repository is https://git.ruby-lang.org built by cgit. We can keep your commit hash on the ruby repository.

Reviewer: ```suggestion
The new repository is https://git.ruby-lang.org built by cgit. We can keep your commit hash on the ruby repository directly.
```

## Development policy

> * We don't use a topic branch on cgit.

Reviewer 2: ```suggestion
* We don't use a topic branch on cgit until Ruby 2.7 is released. ruby_2_7 will use cgit.
```

> * We don't use a topic branch on cgit until Ruby 2.7 is released. ruby_2_7 will use cgit.

Reviewer 2: Oops sorry, I misunderstood the original sentence by myself (topic branch is completely different from stable branch...). Let's rollback that part and note about ruby_2_7 in the later sentence...

```suggestion
* We don't use a topic branch on cgit.
```

* GitHub repository will still be just a mirror. We don't use "Merge pull request" feature.

> * ruby_2_4, ruby_2_5, ruby_2_6 branches will continue to use SVN. We don't push anything to these branches on cgit.

Reviewer 2: ```suggestion
* ruby_2_4, ruby_2_5, ruby_2_6 branches will continue to use SVN. We don't push anything to these branches on cgit.
  * From ruby_2_7, we'll use cgit to develop stable branches.
```

## Special Thanks

* k0kubun

> k0kubun aggresively develop toolchains related release and backport workflow and also update the hook script for git.

Reviewer: ```suggestion
k0kubun aggressively develops toolchains related release and backport workflow and also updates the hook script for git.
```

* naruse

> naruse update the feature changes for Ruby CI and Redmine that is bugs.ruby-lang.org.

Reviewer: ```suggestion
naruse updates the feature changes for Ruby CI and Redmine that is bugs.ruby-lang.org.
```

* mame

> mame create the commit notification script for slack.

Reviewer: ```suggestion
mame creates the commit notification script for slack.
```

---
layout: news_post
title: "Move to Git from Subversion"
author: "hsbt"
translator:
date: 2019-04-23 00:00:00 +0000
lang: en
---

Today, The canonical repository of the Ruby language was moved to Git from Subversion.

> The new repository is https://git.ruby-lang.org built by cgit. We can keep your commit hash on the ruby repository directly.

Reviewer 3: ```suggestion
The web interface for the new repository is [https://git.ruby-lang.org](https://git.ruby-lang.org), and is provided by cgit. We can keep your commit hash on the Ruby repository directly.
```

_note: the given URL is not the git repository itself, therefore the suggested change_

The second sentence is rather unclear: „can“ or „will“?, who is „you“? (every post reader??? ... possibly the existing committers?), what‘s the „commit hash“? ...

## Development policy

* We don't use a topic branch on cgit.

> * GitHub repository will still be just a mirror. We don't use "Merge pull request" feature.

Reviewer 3: ```suggestion
* The GitHub repository will still be just a mirror. We don't use the "Merge pull request" feature.
```

> * ruby_2_4, ruby_2_5, ruby_2_6 branches will continue to use SVN. We don't push anything to these branches on cgit.

Reviewer 3: ```suggestion
* The ruby_2_4, ruby_2_5, and ruby_2_6 branches will continue to use SVN. We don't push anything to these branches on cgit.
```

>   * From ruby_2_7, we'll use cgit to develop stable branches.

Reviewer 3: ```suggestion
* Starting with ruby_2_7, we'll use cgit to develop stable branches.
```

## Special Thanks

* k0kubun

> k0kubun aggressively develops toolchains related release and backport workflow and also updates the hook script for git.

Reviewer 3: ```suggestion
  k0kubun aggressively develops toolchains related to release and backport workflows and also updates the hook script for git.
```

* naruse

> naruse updates the feature changes for Ruby CI and Redmine that is bugs.ruby-lang.org.

Reviewer 3: ```suggestion
  naruse updates the feature changes for Ruby CI and Redmine (bugs.ruby-lang.org).
```

* mame

> mame creates the commit notification script for slack.

Reviewer 3: ```suggestion
  mame creates the commit notification script for slack.
```

> ## Future Works

Reviewer 3: ```suggestion
## Future Work
```

> We still have some of tasks. If you found the issues related git migration, Please file it to https://bugs.ruby-lang.org/issues/14632.

Reviewer 3: ```suggestion
We still have to complete some tasks. If you find an issue related to the git migration, please file it to [https://bugs.ruby-lang.org/issues/14632](https://bugs.ruby-lang.org/issues/14632).
```

---
layout: news_post
title: "Ruby Repository Moved to Git from Subversion"
author: "hsbt"
translator:
date: 2019-04-23 00:00:00 +0000
lang: en
---

Today, the canonical repository of the Ruby programming language was moved to Git from Subversion.

> The web interface for the new repository is [https://git.ruby-lang.org](https://git.ruby-lang.org), and is provided by cgit. We can keep the commit hash from the contributor on the Ruby repository directly.

Reviewer 3: @hsbt 2nd paragraph is still very cryptic. Do you mean something like „Contributers to the SVN repository will also get commit right for the new repository.“ ???

Reviewer 4: I don't get this, too. "Keep something from somebody" is to avoid telling somebody something. But as I read, the meaning of the sentence is more like "We can show the commit hash made by the contributor on the Ruby repository directly."

Reviewer 3: I doubt the Git commit hash is what is meant. I would rather think „commit bit“ or „commit access“.
